### ISSUE TITLE
Mono 4.8.0.495 Breaks TorrentLeech

### ISSUE THREAD

**Issue by llawwehttam** [Jackett/Jackett#1106]
Jackett version:  0.7.1079.0
Mono version (if not using Windows):  4.8.0.495
-------------------------------
I've recently run updates on Ubuntu 16.04.2 which has updated Mono to 4.8.0.495.
After this update Jackett is no longer able to connect to TorrentLeech.

The WebUI shows: "Error - Exception in test_indexer"

I have tried removing and re-adding TorrentLeech which shows "Configuration Failed".

Full logs are as follows:
```
Feb 26 00:19:28 APP-03 mono[193]: 02-26 00:19:28 Debug updating Cookies __cfduid=<snip>; PHPSESSID=<snip>; member_id=<snip>; pas
Feb 26 00:19:28 APP-03 mono[193]: 02-26 00:19:28 Debug Saving new config file: /opt/jackett/.config/Jackett/Indexers/torrentleech.json.<snip>.tmp
Feb 26 00:19:28 APP-03 mono[193]: 02-26 00:19:28 Debug IWebClient.GetString(Url:https://www.torrentleech.org/user/account/login/)
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Debug IWebClient: Returning Redirect => https://www.torrentleech.org/user/account/login/
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Debug IWebClient.GetBytes(Url:https://www.torrentleech.org/user/account/login/)
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Debug IWebClient: Returning OK => 11641 bytes
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Error Exception in test_indexer Jackett.ExceptionWithConfigData
Feb 26 00:19:29 APP-03 mono[193]:   at Jackett.Indexers.TorrentLeech+<DoLogin>c__async1+<DoLogin>c__AnonStorey3.<>m__0 () [0x00022] in C:\projects\jackett\src\Jackett\Indexers\TorrentLe
Feb 26 00:19:29 APP-03 mono[193]:   at Jackett.Indexers.BaseIndexer+<ConfigureIfOK>c__asyncC.MoveNext () [0x00059] in C:\projects\jackett\src\Jackett\Indexers\BaseIndexer.cs:545
Feb 26 00:19:29 APP-03 mono[193]: --- End of stack trace from previous location where exception was thrown ---
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e44
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at Jackett.Indexers.TorrentLeech+<DoLogin>c__async1.MoveNext () [0x00108] in C:\projects\jackett\src\Jackett\Indexers\TorrentLeech.cs:112
Feb 26 00:19:29 APP-03 mono[193]: --- End of stack trace from previous location where exception was thrown ---
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e44
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at Jackett.Indexers.TorrentLeech+<PerformQuery>c__async2.MoveNext () [0x00204] in C:\projects\jackett\src\Jackett\Indexers\TorrentLeech.cs:154
Feb 26 00:19:29 APP-03 mono[193]: --- End of stack trace from previous location where exception was thrown ---
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e44
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at Jackett.Services.IndexerManagerService+<TestIndexer>c__async0.MoveNext () [0x0004f] in C:\projects\jackett\src\Jackett\Services\IndexerManagerServ
Feb 26 00:19:29 APP-03 mono[193]: --- End of stack trace from previous location where exception was thrown ---
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e44
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0
Feb 26 00:19:29 APP-03 mono[193]:   at Jackett.Controllers.AdminController+<Test>c__async5.MoveNext () [0x000da] in C:\projects\jackett\src\Jackett\Controllers\AdminController.cs:269
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Debug Saving new config file: /opt/jackett/.config/Jackett/Indexers/torrentleech.json.9dfbdd44dc2846eb9830304ee5b3bfcd.tmp
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Info ReflectedHttpActionDescriptor ExecuteAsync Action returned 'System.Web.Http.Results.JsonResult`1[Newtonsoft.Json.Linq.JToken]'
Feb 26 00:19:29 APP-03 mono[193]: System.Web.Http.Action: ReflectedHttpActionDescriptor ExecuteAsync Action returned 'System.Web.Http.Results.JsonResult`1[Newtonsoft.Json.Linq.JToken]'
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Info ApiControllerActionInvoker InvokeActionAsync
Feb 26 00:19:29 APP-03 mono[193]: System.Web.Http.Action: ApiControllerActionInvoker InvokeActionAsync
Feb 26 00:19:29 APP-03 mono[193]: 02-26 00:19:29 Info JackettAPINoCacheAttribute OnActionExecutedAsync Action filter for 'Test()'
Feb 26 00:19:29 APP-03 mono[193]: System.Web.Http.Filters: JackettAPINoCacheAttribute OnActionExecutedAsync Action filter for 'Test()'
```

For the time being I've rolled back to Mono 4.6.2.16 which doesn't appear to have any problems.

Is this a known issue with mono 4.8.0.495?

**Comment by noorbertt** [comment]
same... But for some reason only with private torrent pages.

**Comment by flightlevel** [comment]
@noorbertt This happening to you on trackers apart from TorrentLeech?

Am on mobile, so can't test, but really doubt this is related to Mono 4.8 as I've been running that on Ubuntu successfully for a few weeks now. I'd guess its due to this change https://github.com/Jackett/Jackett/commit/3b3420086fa939c1de3a4c7a7fdf46220433dce8

If you try running 1079 with libcurl eg. `mono JackettConsole.exe --UseClient libcurl`
or if you try 1072 and it runs successfully on Mono 4.8, that would confirm that Mono 4.8 isn't the issue

**Comment by llawwehttam** [comment]
Cheers @flightlevel 

I've done some testing and can confirm that version 1079 works for me under Mono 4.8.0.495 when using  `--UseClient libcurl`

For the time being I've added that to my systemd unit file.

I have also tested version 1079 under Mono 4.6.2.16 which appears to work although I'm not sure why.

**Comment by noorbertt** [comment]
--UseClient libcurl its worked, thank you

**Comment by kaso17** [comment]
Fixed with 4fbfc4e7951834e4425846f469175eda21e862b0. With v0.7.1084 you should be able to use httpclient (faster than libcurl) again.
